File: Backend/CIT-Forums/src/main/java/edu/cit/citforums/dto/ThreadDto.java
package edu.cit.citforums.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ThreadDto {
    private Long id;
    private String title;
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Long forumId;
    private String forumTitle;
    private MemberSummaryDto createdBy;
    private int postCount;
    private int viewCount;
    private boolean pinned;
    private boolean locked;
} 

File: MobileDev/app/src/main/java/edu/cit/cituforumsmobile/api/dto/ThreadDto.kt
package edu.cit.cituforumsmobile.api.dto

data class ThreadDto(
    val id: Long,
    val title: String,
    val content: String,
    val createdBy: MemberSummaryDto,
    val forumId: Long,
    val createdAt: String //  Or consider using a Date/Time library
)